src/Crypto/RNCryptor/Types.hs

```diff
@@ -70,9 +73,21 @@ randomSaltIO :: Int -> IO ByteString
 randomSaltIO sz = C8.pack <$> forM [1 .. sz] (const $ randomRIO ('\NUL', '\255'))
 
 --------------------------------------------------------------------------------
+makeKey :: ByteString -> ByteString -> ByteString
+makeKey = generate (prfHMAC (undefined::SHA1)) (Parameters 10000 32) -- userKey salt
+
+--------------------------------------------------------------------------------
+makeHMAC :: ByteString -> ByteString -> ByteString -> ByteString
+makeHMAC hmacSalt userKey secret =
+  let key        = makeKey userKey hmacSalt::ByteString
+      hmacSha256 = hmac key secret::HMAC SHA256
```

Comment:
... but taking that type annotation off of secret causes a compiler error:

    No instance for (HashAlgorithm a0) arising from a use of ‘hmac’
    The type variable ‘a0’ is ambiguous
    Relevant bindings include
      hmacSha256 :: HMAC a0 (bound at src/Crypto/RNCryptor/Types.hs:83:7)
    Note: there are several potential instances:
      instance HashAlgorithm SHA1
        -- Defined in ‘cryptonite-0.15:Crypto.Hash.SHA1’
      instance HashAlgorithm SHA256
        -- Defined in ‘cryptonite-0.15:Crypto.Hash.SHA256’
    In the expression: hmac key secret
    In an equation for ‘hmacSha256’: hmacSha256 = hmac key secret
    In the expression:
      let
        key = makeKey userKey hmacSalt
        hmacSha256 = hmac key secret
      in convert hmacSha256

Much of the cryptonite API is so densely abstracted to type classes that it seems sometimes hints are required?

It's a bit of a mystery to me.